aten/src/ATen/native/cuda/Nonzero.cu

```diff
@@ -216,4 +325,56 @@ Tensor nonzero_cuda(const Tensor& self) {
   Tensor out = at::detail::empty_cuda({0}, self.options().dtype(kLong));
   return at::native::nonzero_out_cuda(self, out);
 }
+
+Tensor& nonzero_static_out_cuda(
+    const Tensor& self,
+    int64_t size,
+    int64_t fill_value,
+    Tensor& out) {
+  TORCH_CHECK(
+      out.dtype() == at::kLong,
+      "nonzero_static: Expected out tensor to have scalar type ",
+      at::kLong,
+      " but got ",
+      out.dtype());
+  TORCH_CHECK(
+      self.device() == out.device(),
+      "expected self and out to be on the same device, but got out on ",
+      out.device(),
+      " and self on ",
+      self.device());
+  TORCH_CHECK(
+      self.dim() <= MAX_DIMS,
+      "nonzero_static is not supported for tensor with more than ",
+      MAX_DIMS,
+      " dimensions");
+  TORCH_CHECK(
+      size >= 0, "nonzero_static: 'size' must be an non-negative integer"
+  )
+  AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND4(
+      at::ScalarType::ComplexHalf,
+      at::ScalarType::Bool,
+      at::ScalarType::BFloat16,
+      at::ScalarType::Half,
+      self.scalar_type(),
+      "nonzero_cuda",
+      [&] {
+        nonzero_static_cuda_out_impl<scalar_t>(self, size, fill_value, out);
+      });
+  return out;
+}
+
+Tensor nonzero_static_cuda(
+    const Tensor& self,
+    int64_t size,
+    int64_t fill_value) {
+  TORCH_CHECK(
+      size >= 0, "nonzero_static: 'size' must be an non-negative integer"
+  )
+  Tensor out = Tensor(at::detail::empty_cuda(
```

Comment:
It's weird you have to wrap this in Tensor before assigning, shouldn't be necessary I think?

Reply:
Copy paste from nonzero impl, I'll check if I can remove both wrappers

Reply:
`error: class "at::TensorBase" has no member "t"`

Reply:
oh lol, okey